discovery-provider/src/tasks/index_rewards_manager.py

```diff
@@ -0,0 +1,369 @@
+    {"name": "id", "type": SolanaInstructionType.string},
+    {"name": "eth_recipient", "type": SolanaInstructionType.EthereumAddress},
+]
+
+
+class RewardsManagerTransfer(TypedDict):
+    amount: int
+    id: str
+    eth_recipient: str
+
+
+def parse_transfer_instruction_data(data: str) -> RewardsManagerTransfer:
+    """Parse Trasfer instruction data submitted to Audius Rewards Manager program
+
+    Instruction struct:
+    pub struct TransferArgs {
+        /// Amount to transfer
+        pub amount: u64,
+        /// ID generated on backend
+        pub id: String,
+        /// Recipient's Eth address
+        pub eth_recipient: EthereumAddress,
+    }
+
+    Decodes the data and parses each param into the correct type
+    """
+
+    return parse_instruction_data(data, rewards_manager_transfer_instr)
+
+
+def parse_transfer_instruction_id(transfer_id: str) -> List[str]:
+    """Parses the transfer instruction id into [challenge_id, specifier]
+    The id in the transfer instruction is formatted as "<CHALLENGE_ID>:<SPECIFIER>"
+    """
+    id_parts = transfer_id.split(":")
+    if len(id_parts) != 2:
+        raise Exception(
+            f"Unable to parse tranfer instruction id into challenge_id and specifier {transfer_id}"
+        )
+    return id_parts
+
+
+def get_valid_instruction(
+    tx_message: TransactionMessage, meta: ResultMeta
+) -> Optional[TransactionMessageInstruction]:
+    """Checks the transaction message for a transfer instruction and validates it"""
+    instructions = tx_message["instructions"]
+    rewards_manager_program_index = tx_message["accountKeys"].index(
+        REWARDS_MANAGER_PROGRAM
+    )
+    has_transfer_instruction = any(
+        log == "Program log: Instruction: Transfer" for log in meta["logMessages"]
+    )
+    if has_transfer_instruction:
+        for instruction in instructions:
+            if instruction["programIdIndex"] == rewards_manager_program_index:
+                return instruction
+
+    return None
+
+
+def process_sol_rewards_transfer_instruction(
+    session: Session, solana_client: Client, tx_sig: str
+):
+    """Fetches metadata for rewards tranfer transactions and creates Challege Disbursements
+
+    Fetches the transaction metadata from solana using the tx signature
+    Checks the metadata for a transfer instruction
+    Decodes and parses the transfer instruction metadata
+    Validates the metadata fields and inserts a ChallengeDisbursement DB entry
+    """
+    try:
+        tx_info = get_sol_tx_info(solana_client, tx_sig)
+        result: TransactionInfoResult = tx_info["result"]
+        meta = result["meta"]
+        if meta["err"]:
+            logger.info(
+                f"index_rewards_manager.py | Skipping error transaction from chain {tx_info}"
+            )
+            return
+        tx_message = result["transaction"]["message"]
+        instruction = get_valid_instruction(tx_message, meta)
+        if instruction is not None:
```

Comment:
nit we could just invert this check and return early to avoid nesting the rest of the happy path